site/src/pages/TemplatePage/TemplateInsightsPage/IntervalMenu.tsx

```diff
@@ -0,0 +1,87 @@
+import CheckOutlined from "@mui/icons-material/CheckOutlined";
+import ExpandMoreOutlined from "@mui/icons-material/ExpandMoreOutlined";
+import Box from "@mui/material/Box";
+import Button from "@mui/material/Button";
+import Menu from "@mui/material/Menu";
+import MenuItem from "@mui/material/MenuItem";
+import { useState, useRef } from "react";
+
+export const insightsIntervals = {
+  day: {
+    label: "Daily",
+  },
+  week: {
+    label: "Weekly",
+  },
+} as const;
+
+export type InsightsInterval = keyof typeof insightsIntervals;
+
+export const IntervalMenu = ({
+  value,
+  onChange,
+}: {
+  value: InsightsInterval;
+  onChange: (value: InsightsInterval) => void;
+}) => {
+  const anchorRef = useRef<HTMLButtonElement>(null);
+  const [open, setOpen] = useState(false);
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  return (
+    <div>
+      <Button
+        ref={anchorRef}
+        id="interval-button"
+        aria-controls={open ? "interval-menu" : undefined}
+        aria-haspopup="true"
+        aria-expanded={open ? "true" : undefined}
+        onClick={() => setOpen(true)}
+        endIcon={<ExpandMoreOutlined />}
+      >
+        {insightsIntervals[value].label}
+      </Button>
+      <Menu
+        id="interval-menu"
+        anchorEl={anchorRef.current}
+        open={open}
+        onClose={handleClose}
+        MenuListProps={{
+          "aria-labelledby": "interval-button",
+        }}
+        anchorOrigin={{
+          vertical: "bottom",
+          horizontal: "left",
+        }}
+        transformOrigin={{
+          vertical: "top",
+          horizontal: "left",
+        }}
+      >
+        {Object.keys(insightsIntervals).map((interval) => {
```

Comment:
I'd have to double-check, but I want to make sure that `Object.keys` doesn't guarantee a consistent order for the array it gives you, just at the language specification level (even though Chrome's really good about it)

Do you think it'd be overkill to sort the keys, just to make sure that the items don't randomly jump around each render?

Reply:
Updated!